### PR TITLE
Add RSS feed

### PR DESCRIPTION
**Pull request by milo**
I tried to keep app style. Not sure about feed URL.

### CODE REVIEW

**Comment by f3l1x** [comment]
Thank you. :+1: 

Style is great. Url is oukey.

I was thinking about RssControl, it could be more independend on Addons. For now, you have only newest, but what about newest, updated and all?

There could be a RssControlFactory with 3 methods `createNewest()`, `createUpdated()`, `createAll()`. 

And one little thing, cache could be at least for +1 hour, that's cron interval. :)

What do you think?

**Comment by milo** [comment]
Sure. I started with the most simplest one. I'll update the PR later.

**Comment by milo** [comment]
@f3l1x When thinking about it, I'm not sure it is good idea for `updated` (and `all`) feeds. Which information would you put into the article body on update (I suppose https://componette.com/all/?search-by=push). There is no diff to show.

**Comment by f3l1x** [comment]
@milo There's no change to track changes with 25 limit. ATM we have over 280 addons, at RSS will be last 25.

So, what about newest and all (with cache +1 day)?

**Comment by milo** [comment]
And what will be in `all` feed? All 280 addons? Seems strange to. Imho, newest is the only one reasonable feed.

File: app/modules/front/controls/RssFeed/RssFeed.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @return array
+     */
+    private function getItems()
+    {
+        $items = [];
+        foreach ($this->facade->findNewest(25) as $addon) {
+            $items[] = (object) [
+                'guid' => "$addon->id@componette.com",
+                'title' => "[$addon->type] $addon->owner/$addon->name",
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/RssFeed.php]: `$addon->owner/$addon->name` -> `$addon->fullname`

File: app/modules/front/controls/RssFeed/RssFeed.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * RENDER ******************************************************************
+     */
+
+    public function render()
+    {
+        $this->template->title = 'Addons on Componette';
+        $this->template->link = $this->presenter->link('//:Front:Home:default');
+        $this->template->description = 'List of new addons as added by users on Componette.';
+        $this->template->time = new \DateTime;
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/RssFeed.php]: I preffer Nette\Utils\DateTime with `use`.

File: app/modules/front/controls/RssFeed/templates/rss-feed.latte [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		<link>{$link}</link>
+		<atom:link rel="self" href="{$link}" type="application/rss+xml" />
+		<language>en-us</language>
+		<description>{$description}</description>
+		<pubDate>{$time |date:\DateTime::RSS}</pubDate>
+		<item n:foreach="$items as $item">
+			{capture $html}{$item->content|parsedown|noescape}{/}
+			<guid isPermaLink="false">{$item->guid}</guid>
+			<title>{$item->title}</title>
+			<link>{$item->link}</link>
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/templates/rss-feed.latte]: Add <pubDate> also to <item>, you can use `$addon->createdAt`

**Comment by f3l1x** [comment]
I found 3 little corrections, it's fine other. :)

**Comment by milo** [comment]
I updated 2 more things:
- dates to UTC
- noreply email in <autor> tag (due to validity)

Let me know when ready to squash.

File: app/modules/front/controls/RssFeed/templates/rss-feed.latte [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		<description>{$description}</description>
+		<pubDate>{$time|date:\DateTime::RSS}</pubDate>
+		<item n:foreach="$items as $item">
+			{capture $html}{$item->content|parsedown|noescape}{/}
+			<guid isPermaLink="false">{$item->guid}</guid>
+			<title>{$item->title}</title>
+			<link>{$item->link}</link>
+			<pubDate>{$item->time|date:\DateTime::RSS}</pubDate>
+			<author>{$item->author}</author>
+			<description>{$html |stripTags}</description>
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/templates/rss-feed.latte]: {$html|striptags}

File: app/modules/front/controls/RssFeed/RssFeed.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * RENDER ******************************************************************
+     */
+
+    public function render()
+    {
+        $this->template->title = 'Componette - new addons';
+        $this->template->link = $this->presenter->link('//:Front:Home:default');
+        $this->template->description = 'List of new addons as added by users on Componette.';
+        $this->template->time = (new DateTime)->setTimezone(new \DateTimeZone('UTC'));
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/RssFeed.php]: I prefer  using:
```php
use DateTimeZone;
```

File: app/modules/front/controls/RssFeed/RssFeed.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    private function getItems()
+    {
+        $items = [];
+        foreach ($this->facade->findNewest(25) as $addon) {
+            $items[] = (object) [
+                'guid' => "$addon->id@componette.com",
+                'title' => "[$addon->type] $addon->fullname",
+                'link' => $this->presenter->link('//:Front:Addon:detail', ['slug' => $addon->id]),
+                'time' => $addon->createdAt->setTimezone(new \DateTimeZone('UTC')),
```
Comment (f3l1x) [inline app/modules/front/controls/RssFeed/RssFeed.php]: I prefer  using:
```php
use DateTimeZone;
```

**Comment by f3l1x** [comment]
After these 3 things you can squash it to 1 commit. And I'll merge it asap.

**Comment by milo** [comment]
@f3l1x Fixed and squashed.

**Comment by f3l1x** [comment]
@milo Thanks a lot. Can you please also rebase to master?

**Comment by milo** [comment]
@f3l1x Done

**Comment by f3l1x** [comment]
@milo Thank you. :) :+1:

**Comment by milo** [comment]
@f3l1x Thank you! When is planned new deploy? :)

**Comment by f3l1x** [comment]
@milo I think tomorrow.

**Pull request merged**